src/components/widgets/components/amount-input/amount-input.tsx

```diff
@@ -0,0 +1,146 @@
+import BigNumber from 'bignumber.js';
+import { Blockchain } from '../../../../core/blockchain/types';
+import { getSelectedBlockchain } from '../../../../redux/wallets/selectors';
+import { getBlockchain } from '../../../../core/blockchain/blockchain-factory';
+import { IScreenInputDataValidations } from '../../../../redux/ui/screens/input-data/state';
+
+interface IExternalProps {
+    module: IScreenModule;
+    context: IScreenContext;
+    actions: {
+        runScreenValidation?: typeof runScreenValidation;
+    };
+    screenValidation: IScreenValidation;
+}
+
+interface IReduxProps {
+    blockchain: Blockchain;
+
+    inputAmount: string;
+    screenAmount: string;
+    inputValidation: IScreenInputDataValidations;
+
+    setScreenInputData: typeof setScreenInputData;
+    setScreenAmount: typeof setScreenAmount;
+}
+
+const mapStateToProps = (state: IReduxState, ownProps: IExternalProps) => {
+    const flowId = ownProps.context?.flowId;
+
+    return {
+        blockchain: getSelectedBlockchain(state),
+
+        inputAmount: flowId && state.ui.screens.inputData[flowId]?.data?.inputAmount,
+        screenAmount: flowId && state.ui.screens.inputData[flowId]?.data?.screenAmount,
+        inputValidation: flowId && state.ui.screens.inputData[flowId]?.validation
+    };
+};
+
+const mapDispatchToProps = {
+    setScreenInputData,
+    setScreenAmount
+};
+
+class AmountInputComponent extends React.Component<
+    IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>> & IExternalProps
+> {
+    public componentDidMount() {
+        if (this.props.context?.flowId) {
+            this.props.setScreenAmount(this.props.context.flowId);
+        }
+    }
+
+    public render() {
+        const {
+            inputAmount,
+            blockchain,
+            context,
+            module,
+            inputValidation,
+            screenAmount,
+            styles,
+            theme
+        } = this.props;
+
+        const { flowId } = context;
+
+        const data = module.data as IAmountInputData;
+
+        const formattedScreenAmount = formatNumber(new BigNumber(screenAmount || '0'), {
+            currency: getBlockchain(blockchain).config.coin
+        });
+
+        return (
+            <View style={[styles.container, formatStyles(module?.style)]}>
+                <View style={[styles.inputBox, formatStyles(data?.input?.style)]}>
+                    <TextInput
+                        testID="enter-amount"
+                        style={styles.inputText}
+                        autoCapitalize={'none'}
+                        autoCorrect={false}
+                        selectionColor={theme.colors.accent}
+                        value={inputAmount}
+                        onChangeText={text => {
+                            text = text.replace(/,/g, '.');
+                            this.props.setScreenInputData(flowId, text, 'inputAmount');
+                            this.props.actions.runScreenValidation(
```

Comment:
you can call this action inside `setScreenInputData`, you have there everything you need.